telemetry/device_id.py

```
@@ -0,0 +1,97 @@
+import re
+import socket
+import subprocess
+
+
+def _sha256(text: str) -> str:
+    return hashlib.sha256(text.encode()).hexdigest()
+
+
+def _run(cmd: str) -> str:
+    """Mirror Node's ``execSync(cmd).toString()`` (shell, stdout as text)."""
+    return subprocess.check_output(cmd, shell=True, text=True, stderr=subprocess.DEVNULL)
```

Ruff (0.15.15)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)